src/Combinatorics/Matroids/quantum_automorphism_groups.jl

```diff
@@ -0,0 +1,175 @@
+@doc raw"""
+    quantum_symmetric_group(n::Int)
+
+Return the ideal that defines the quantum symmetric group on `n` elements.
+It is comprised of `2*n + n^2 + 2*n*n*(n-1)` many generators.
+
+The relations are:
+
+  - row and column sum relations: `2*n` relations
+  - idempotent relations: `n^2` relations
+  - relations of type `u[i,j]*u[i,k]` and `u[j,i]*u[k,i]` for `k != j`: `2*n*n*(n-1)` relations
+
+# Example
+
+```jldoctest
+julia> S4 = quantum_symmetric_group(4);
+
+julia> length(gens(S4))
+120
+```
+"""
+function quantum_symmetric_group(n::Int)
+  A, u = free_associative_algebra(QQ, :u => (1:n, 1:n))
+
+  relations = elem_type(A)[]
+
+  #Idempotent relations
+  for i in 1:n, j in 1:n
+    new_relation = u[i,j] * u[i,j] - u[i,j]
+    push!(relations, new_relation)
+    for k in 1:n
+      if k != j
+        new_relation = u[i,j] * u[i,k]
+        push!(relations, new_relation)
+        new_relation = u[j,i] * u[k,i]
+        push!(relations, new_relation)
+      end
+    end
+  end
+
+  #row and column sum relations
+  for i in 1:n
+    push!(relations, sum(u[i,:]) - 1)
+    push!(relations, sum(u[:,i]) - 1)
+
+  end
+  return ideal(relations)
+end
+
+@doc raw"""
+    _quantum_automorphism_group_indices(M::Matroid, structure::Symbol=:bases)
+
+Return the indices of the relations that define the quantum automorphism group of a matroid for a given structure.
+
+# Examples
+
+```jldoctest
+julia> M = uniform_matroid(3,4);
+
+julia> idx = Oscar._quantum_automorphism_group_indices(M,:bases);
+
+julia> length(idx)
+1920
+```
+"""
+function _quantum_automorphism_group_indices(M::Matroid, structure::Symbol=:bases)
+  @req structure in [:bases, :circuits, :flats] "structure must be one of :bases, :circuits, :flats"
+
+  n = length(M)
+  grdSet = matroid_groundset(M)
+
+  b    = [Vector{Int}[] for _ in 1:n]
+  nb   = [Vector{Int}[] for _ in 1:n]
+  rels = [Vector{Tuple{Int,Int}}[] for _ in 1:n]
+
+  sets  = getproperty(Oscar, structure)(M)
+  sizes = unique!(map(length, sets))
+  tempGrdSet = reduce(vcat,[grdSet for i in 1:n])
+
+  for size in sizes 
+    size == 0 && continue
+    powerSet = unique(sort.(subsets(tempGrdSet,size)))
+
+    setsOfSize = filter(x->length(x)==size,sets)
+    nonSets = setdiff(powerSet,setsOfSize) 
+
+    for set in setsOfSize
+      append!(b[size],collect(permutations(set)))
```

Comment:
the function `permutations` used here is from the `experimental` folder and should thus not be used in code in `src`. @fingolfin do you by chance  know the syntax how to do this properly?

Reply:
I added an alternative way. Lets hope the test succeed now.